chore(l2): update prover docs (#5105)
lambdaclass/ethrex: docs/developers/l2/prover.md

# Running the Prover

This guide provides instructions for setting up and running the Ethrex L2 prover for development and testing purposes.

## Dependencies

Before you begin, ensure you have the following dependencies installed:

- [RISC0](https://dev.risczero.com/api/zkvm/install)
  1. `curl -L https://risczero.com/install | bash`
  2. `rzup install`
- [SP1](https://docs.succinct.xyz/docs/sp1/introduction)
  1. `curl -L https://sp1up.succinct.xyz | bash`
  2. `sp1up --version 5.0.8`
- [SOLC](https://docs.soliditylang.org/en/latest/installing-solidity.html)

After installing the toolchains, a quick test can be performed to check if we have everything installed correctly.

### L1 block proving

ethrex-prover is able to generate execution proofs of Ethereum Mainnet/Testnet blocks. An example binary was created for this purpose in `crates/l2/prover/bench`. Refer to its README for usage.

## Dev Mode

> To run the blockchain (`proposer`) and prover in conjunction, start the `Prover`, use the following command:

Reviewer: I think the sentence is more clear with something like this
```suggestion
To run the blockchain (`proposer`) and prover in conjunction by starting the `Prover` with:
```

      ETHREX_PROOF_COORDINATOR_L1_PRIVATE_KEY=<private_key>
      # Used to handle TCP communication with other servers from any network interface.
      ETHREX_PROOF_COORDINATOR_LISTEN_ADDRESS=0.0.0.0
      # Set to true to randomize the salt.
      ETHREX_DEPLOYER_RANDOMIZE_CONTRACT_DEPLOYMENT=true
      # Set to true if you want SP1 proofs to be required
      ETHREX_L2_SP1=true
      # Check the if the verification contract is present on your preferred network. Don't define this if you want it to be deployed automatically.
      ETHREX_DEPLOYER_SP1_VERIFIER_ADDRESS=<address>
      # Set to true if you want proofs to be required
      ETHREX_L2_RISC0=true
      # Check the if the contract is present on your preferred network. You shall deploy it manually if not.
      ETHREX_DEPLOYER_RISC0_VERIFIER_ADDRESS=<address>
      # Set to any L1 endpoint.
      ETHREX_ETH_RPC_URL=<url>
      ```
   3. `source .env`

> [!NOTE]
> Make sure to have funds, if you want to perform a quick test `0.2[ether]` on each account should be enough.

- `Finally`, to start the `proposer`/`l2 node`, run:
  - `make rm-db-l2 && make down`

>   - `make deploy-l1 && make init-l2` (if running a risc0 prover, see the next step before invoking the L1 contract deployer)

Reviewer: ```suggestion
  - `make rm-db-l2 down`
  - `make deploy-l1 init-l2` (if running a risc0 prover, see the next step before invoking the L1 contract deployer)
```
